### ISSUE TITLE
Performance web API implementation crashing Safari 12/iOS 14/Android 9/Firefox 102 and earlier

### ISSUE THREAD

**Issue by MaoSHYJ** [vercel/next.js#54389]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:19 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T8103
    Binaries:
      Node: 20.4.0
      npm: 7.24.2
      Yarn: 1.22.19
      pnpm: 8.6.12
    Relevant Packages:
      next: 13.4.19
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 4.5.5
    Next.js Config:
      output: standalone
```

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/vercel/next.js/tree/v13.4.17

### To Reproduce

Any project that is in at least v13.4.17 on safari 12 will crash

### Describe the Bug

On https://github.com/vercel/next.js/blob/v13.4.17/packages/next/src/client/index.tsx#L457
You can see that it is using PerformanceMeasure return interface, which is not implemented on Safari 12

### Expected Behavior

To not crash on Safari 12

### Which browser are you using? (if relevant)

Safari 12.1.2

**Comment by apuyou** [comment]
I am facing the same issue, the error collected by Sentry is `undefined is not an object (evaluating 'e.startTime')` and happens on older devices only (iOS 14/Android 9 and earlier).
 
This started in 13.4.17 and seems related to https://github.com/vercel/next.js/pull/54078

**Comment by WavyWalk** [comment]
We faced the same issue after updating to 13.4.17.

For us update broke Firefox ~102 and other mentioned browsers in production as well.

**Comment by tavvy** [comment]
Same issues here. Older iOS devices. We've downgraded to `13.4.16`

**Comment by galaxynova1999** [comment]
same here for ios12.6 devices

error msg: TypeError: undefined is not an object (evaluating 'beforeHydrationMeasure.startTime')

**Comment by leeDav** [comment]
Is there any update/recognition of this from the Next.js team?

This should not have been a minor version bump but a major/breaking change, especially given [Next's own supported browser list](https://nextjs.org/docs/architecture/supported-browsers) still cites Safari 12 and up.

Until this is resolved we're stuck on `13.4.16`.
